core/domain/prod_validation_jobs_one_off.py

```diff
@@ -4566,11 +4514,20 @@ def _validate_story_is_public(cls, item):
             # function.
             if story_model is None or story_model.deleted:
                 continue
-            story_rights = story_services.get_story_rights(story_model.id)
-            if not story_rights.story_is_published:
-                cls.errors['public story check'].append(
-                    'Entity id %s: Story with id %s corresponding to entity '
-                    'is private' % (item.id, story_model.id))
+            topic_id = story_model.corresponding_topic_id
+            if topic_id:
+                topic = topic_models.TopicModel.get_by_id(topic_id)
+                all_story_references = (
+                    topic.canonical_story_references +
```

Comment:
Suggest abstracting into topic.get_all_story_references() and using that elsewhere too. (Make sure to add unit tests for that new method.)

Reply:
Done.